camel-core/src/main/java/org/apache/camel/impl/DefaultMessage.java

```diff
@@ -67,6 +68,8 @@ public Object getHeader(String name, Object defaultValue) {
     }
 
     public Object getHeader(String name, Supplier<Object> defaultValueSupplier) {
+        IllegalArgumentAssertion.assertNotNull(name, "name");
+        IllegalArgumentAssertion.assertNotNull(defaultValueSupplier, "defaultValueSupplier");
```

Comment:
@tdiesler can we replace IllegalArgumentAssertion.assertNotNull with ObjectHelper.notNull(type, name) for consistency  across camel ?